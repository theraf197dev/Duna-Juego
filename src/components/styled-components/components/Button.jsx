import React from 'react';
import {
    ButtonStyles,
    ContainerButtonStyles,
} from '../styles/Button.styles';

const Button = ({
    buttonText,
    handleClick,
}) => {
  return (
    <ContainerButtonStyles>
        <ButtonStyles onClick={handleClick}>
            {buttonText}
        </ButtonStyles>
    </ContainerButtonStyles>
  )
};

export default Button;