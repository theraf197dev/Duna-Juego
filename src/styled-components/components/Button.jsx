import React from 'react';
import {
    ButtonStyles,
    ContainerButtonStyles,
    SpanStyles,
} from '../styles/Button.styles';

const Button = ({
    buttonText,
    handleSubmit,
}) => {
  return (
    <ContainerButtonStyles>
        <ButtonStyles onClick={handleSubmit}>
            {buttonText}
            <SpanStyles id={1} />
            <SpanStyles id={2} />
            <SpanStyles id={3} />
            <SpanStyles id={4} />
        </ButtonStyles>
    </ContainerButtonStyles>
  )
}

export default Button