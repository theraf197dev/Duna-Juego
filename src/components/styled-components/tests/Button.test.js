import { render } from "@testing-library/react";
import Button from '../components/Button.jsx';

describe('Button tests', () => {
  let props = {
    buttonText: 'testButton',
    handleClick: jest.fn(),
  };

  it('render component', () => {
      const { container } = render(<Button {...props} />);

      expect(container).toMatchSnapshot();
  });
});
