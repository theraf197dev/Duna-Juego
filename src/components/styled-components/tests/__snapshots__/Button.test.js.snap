// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests render component 1`] = `
<div>
  <div
    class="sc-bcXHqe dQMZft"
  >
    <button
      class="sc-gswNZR kWxGrk"
    >
      testButton
    </button>
  </div>
</div>
`;
